controllers/v1/surveysController.js

```diff
@@ -0,0 +1,824 @@
+
+    constructor() {
+        super(surveysSchema);
+    }
+
+    static get name() {
+        return "surveys";
+    }
+    
+     /**
+     * @api {post} /assessment/api/v1/surveys/createSolutionTemplate Create survey template solution.
+     * @apiVersion 1.0.0
+     * @apiName Create survey solution template.
+     * @apiGroup Surveys
+     * @apiHeader {String} X-authenticated-user-token Authenticity token
+     * @apiSampleRequest /assessment/api/v1/surveys/createSolutionTemplate
+     * @apiParamExample {json} Request-Body:
+     * {
+     *   "name": "Test survey 1",
+     *   "description": "Test survey 1"
+     *   "externalId": "Test-survey",
+     *   "startDate": "2020/10/10"
+     *   "endDate": "2020/12/25",
+     *   "linkTite": "Read more about the solution",
+     *   "linkUrl": "http://www.africau.edu/images/default/sample.pdf"
+     * }
+     * @apiParamExample {json} Response:
+     * {
+         "status": 200,
+         "message": "Survey solution template created successfully",
+         "result": {
+            "solutionId": "5f58b0b8894a0928fc8aa9b3"
+         }
+     * }
+     * @apiUse successBody
+     * @apiUse errorBody
+     */
+
+    /**
+    * Create survey solution template.
+    * @method
+    * @name createSolutionTemplate
+    * @param {Object} req -request Data.
+    * @param {String} req.body- survey solution creation data.  
+    * @returns {JSON} - solutionId
+    */
+
+    async createSolutionTemplate(req) {
+      return new Promise(async (resolve, reject) => {
+        try {
+            
+            let createSolutionTemplate = await surveysHelper.createSolutionTemplate
+            (
+                req.body,
+                req.userDetails.userId
+            )
+
+            return resolve({
+                message: createSolutionTemplate.message,
+                result: createSolutionTemplate.data
+            })
+
+        } catch (error) {
+            return reject({
+                status: error.status || httpStatusCode.internal_server_error.status,
+                message: error.message || httpStatusCode.internal_server_error.message,
+                errorObject: error
+            });
+        }
+    });
+}
+
+
+     /**
+     * @api {post} /assessment/api/v1/surveys/importSurveryTemplateToSolution/:solutionId Import template survey solution to solution.
+     * @apiVersion 1.0.0
+     * @apiName Import survey template to solution.
+     * @apiGroup Surveys
+     * @apiHeader {String} X-authenticated-user-token Authenticity token
+     * @apiSampleRequest /assessment/api/v1/surveys/importSurveryTemplateToSolution/5f58b0b8894a0928fc8aa9b3
+     * @apiParamExample {json} Response:
+     * {
+         "status": 200,
+         "message": "Survey solution imported successfully",
+         "result": {
+            "solutionId": "5f678f1344a7fb63999707d9"
+            "link": "https://apps.shikshalokam.org/samiksha/take-survey/8e8902eb1e13ae9c38dec8f1b5a4bdff"
```

Comment:
@deepakr123 - app name is missing in request to generate link

Reply:
resolved